src/build.sh

```diff
@@ -247,9 +254,21 @@ function full_cleanup()
   cmd_manager "$flag" "$cmd"
 }
 
+function from_sha()
+{
+  local flag="$1"
+  local env_path="$2"
+  local sha="$3"
+  local cmd
+
+  flag=${flag:-'SILENT'}
+  cmd="git rebase ${sha} --exec 'yes \"\" | make prepare${env_path}; make${env_path}'"
```

Comment:
You can't start to run git rebase directly; you first need to check if the repository is in a good state to run it.